assembl/static2/js/app/components/debate/survey/navigation.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { Grid, Col } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { Translate } from 'react-redux-i18n';
+import { getDomElementOffset, scrollToElement, calculatePercentage } from '../../../utils/globalFunctions';
+import { getIfPhaseCompletedByIdentifier } from '../../../utils/timeline';
+
+class Navigation extends React.Component {
+  constructor(props) {
+    super(props);
+    const { questionsLength } = this.props;
+    this.displayNav = this.displayNav.bind(this);
+    this.displayPagination = this.displayPagination.bind(this);
+    this.scrollToNextQuestion = this.scrollToNextQuestion.bind(this);
+    this.scrollToPreviousQuestion = this.scrollToPreviousQuestion.bind(this);
+    this.state = {
+      navPosition: 'fixed',
+      navBottom: 0,
+      isHidden: true,
+      questionsLength: questionsLength,
+      currentQuestionNumber: 0
+    };
+  }
+  componentDidMount() {
+    window.addEventListener('scroll', this.displayNav);
```

Comment:
Please use lodash's debounce or throttle, because otherwise as the browser sends these scroll events very often, running the listener code too often can kill performance and produce lag.